src/components/ui/Input/NumberInput.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * This file is based on the Input component from shadcn and customized for our needs.
+ *
+ * See the official docs for more info:
+ * shadcn/ui: https://ui.shadcn.com/docs/components/input
+ */
+
+'use client';
+
+import { cn } from '@/lib/tailwind/utils';
+
+import { VariantProps } from 'class-variance-authority';
+import { FC, InputHTMLAttributes } from 'react';
+
+import { Input, inputVariants } from './Input';
+
+interface IExceptionSymbolsProps {
+  exceptLeadingZero?: boolean;
+  exceptNegative?: boolean;
+  exceptPlusOperator?: boolean;
+  exceptE?: boolean;
+  exceptDot?: boolean;
+}
+
+interface INumberInputProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type'>,
+    VariantProps<typeof inputVariants>,
+    IExceptionSymbolsProps {}
+
+const zero = '0';
+const negative = '-';
+const plus = '+';
+const e = ['e', 'E'];
+const dot = '.';
+
+const handleExceptionSymbols = (
```

Comment:
The code for this logic might be a bit unnecessaly lengthy... If you come up with better one, please tell me 🙏

Reply:
Shall we have a conversation about this?
I am not even sure what this file is trying to achieve.